src/hypothesis/database.py

```diff
@@ -20,19 +20,48 @@
 import os
 import re
 import binascii
+import warnings
 import threading
 from hashlib import sha1
 from contextlib import contextmanager
 
+from hypothesis.errors import HypothesisWarning
 from hypothesis._settings import note_deprecation
+from hypothesis.configuration import storage_directory
 from hypothesis.internal.compat import FileNotFoundError, hbytes, \
     b64decode, b64encode
+from hypothesis.utils.conventions import infer
 
 sqlite3 = None
 SQLITE_PATH = re.compile(r"\.\(db|sqlite|sqlite3\)$")
 
 
 def _db_for_path(path=None):
+    if path is infer:
+        path = os.getenv('HYPOTHESIS_DATABASE_FILE')
+        if path is not None:  # pragma: no cover
+            # Note: for Hypothesis 4, there should still be a deprecation
+            # warning (noting that the var is ignored) to ease debugging for
```

Comment:
As per discussion before (depending on timeline) this probably won't be true in Hypothesis 4 - we'll still respect the environment variable.